rpm/kubelet.spec

```diff
@@ -0,0 +1,87 @@
+%global KUBE_VERSION 1.3.4
+%global CNI_RELEASE 8a936732094c0941e1543ef5d292a1f4fffa1ac5
+
+Name: kubelet
+Version: %{KUBE_VERSION}
+Release: 1
+Summary: Container cluster management
+License: ASL 2.0
+
+URL: https://kubernetes.io
+Source0: https://storage.googleapis.com/kubernetes-release/release/v%{KUBE_VERSION}/bin/linux/amd64/kubelet
+Source1: kubelet.service
+
+BuildRequires: curl
+Requires: docker-engine >= 1.10
+Requires: iptables >= 1.4.21
+Requires: socat
+Requires: util-linux
+Requires: ethtool
+
+%description
+The node agent of Kubernetes, the container cluster manager.
+
+%package plugin-cni
+
+Summary: Binaries required to provision kubernetes container networking
+Requires: kubelet
+
+%description plugin-cni
+Binaries required to provision container networking.
+
+
+%prep
+# Assumes the builder has overridden sourcedir to point to directory
+# with this spec file. (where these files are stored) Copy them into
+# the builddir so they can be installed.
+#
+# Example: rpmbuild --define "_sourcedir $PWD" -bb kubelet.spec
+#
+cp -p %{_sourcedir}/kubelet.service %{_builddir}/
+cp -p %{_sourcedir}/kubelet-wrapper %{_builddir}/
+cp -p %{_sourcedir}/99_bridge.conf %{_builddir}/
+cp -p %{_sourcedir}/kubelet.env %{_builddir}/
+
+#cp -p %{_sourcedir}/kubelet %{_builddir}/
+
+
+%install
+
+curl -L --fail "https://storage.googleapis.com/kubernetes-release/release/v%{KUBE_VERSION}/bin/linux/amd64/kubelet" -o kubelet
+
+install -m 755 -d %{buildroot}%{_bindir}
+install -m 755 -d %{buildroot}%{_sysconfdir}/systemd/system/
+install -m 755 -d %{buildroot}%{_sysconfdir}/sysconfig
+install -m 755 -d %{buildroot}%{_sysconfdir}/cni/net.d/
+install -m 755 -d %{buildroot}%{_sysconfdir}/kubernetes/manifests/
+install -m 755 -d %{buildroot}/var/lib/kubelet/
+install -p -m 755 -t %{buildroot}%{_bindir}/ kubelet
+install -p -m 755 -t %{buildroot}%{_sysconfdir}/systemd/system/ kubelet.service
+install -p -m 755 -t %{buildroot}%{_sysconfdir}/cni/net.d/ 99_bridge.conf
+install -p -m 755 kubelet.env %{buildroot}%{_sysconfdir}/sysconfig/kubelet
+install -p -m 755 -t %{buildroot}/var/lib/kubelet/ kubelet-wrapper
+
+
+install -m 755 -d %{buildroot}/opt/cni
+curl -sSL --fail --retry 5 https://storage.googleapis.com/kubernetes-release/network-plugins/cni-%{CNI_RELEASE}.tar.gz | tar xz
```

Comment:
This is implemented as a "sub-package" so it's built at the same time, but the output is a separate rpm that would have to be installed separately if the user wanted it. (i.e. just a logical grouping for the build)

Could do a separate spec file for CNI if that is desirable for any reason, not aware of any off hand but just let me know!

For reference the output from running this is:

```
(dgoodwin@wrx {rpms} ~/src/release/rpm/output) $ tree
.
└── x86_64
    ├── kubelet-1.3.4-1.x86_64.rpm
    ├── kubelet-plugin-cni-1.3.4-1.x86_64.rpm
    └── repodata
        ├── 1eabb97862594ab0ecb335b5f05dc965964c1244001d3975b1975dcc94f4cf4f-filelists.sqlite.bz2
        ├── 1f3c7c26284ab502308f6916703fe2962770700b465b664a060146e44e3889b9-other.sqlite.bz2
        ├── 9bdca64537b7200b3f8509bd980c246ff43990a4b809950f05d9a92a2eb49a9f-primary.sqlite.bz2
        ├── eb03fa57fa4ef08a34542efb1ada981c0b6e3d43711eb69217ad70616a29c44a-other.xml.gz
        ├── eb5bb756df3a40bce6bfe5af45592904b7cb4af37fcff1f374cd8b8a53d5d47e-filelists.xml.gz
        ├── f15a5d506258ff07b9a598a4b13f732df3edec5f902c1b09b973a9ff133e8e92-primary.xml.gz
        └── repomd.xml

2 directories, 9 files
```